src/domain/value-objects/Password.ts

```diff
@@ -0,0 +1,120 @@
+import { Either } from "../entities/Either";
+import { validateLengthMax, validateLengthMin, validateRegexp, validateRequired } from "../utils/validations";
+import { ValueObject } from "./ValueObject";
+
+export interface PasswordProps {
+    value: string;
+}
+
+export class Password extends ValueObject<PasswordProps> {
+    public readonly value: string;
+
+    private constructor(props: PasswordProps) {
+        super(props);
+
+        this.value = props.value;
+    }
+
+    public static create(value: string, isExistingUser = false): Either<string[], Password> {
+        // For existing users, empty password is allowed
+        if (isExistingUser && !value) {
+            return Either.success(new Password({ value }));
+        }
+
+        const requiredError = validateRequired(value, "Please provide a password");
+
+        if (requiredError) {
+            return Either.error([requiredError]);
+        }
+
+        const minLengthError = validateLengthMin(value, 8, "Password should be at least 8 characters long");
```

Comment:
Probably out of the scope of this PR, but the minLength value depends on a server setting.

<img width="1305" height="342" alt="image" src="https://github.com/user-attachments/assets/ce8ef171-1fe9-4267-8776-9d9151b6a573" />

Maybe we will need a new Entity `PasswordPolicy` or `PasswordRules`? 

We can get this value from the API `/api/systemSettings/minPasswordLength`

https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-240/settings-and-configuration.html (Minimum length of password)